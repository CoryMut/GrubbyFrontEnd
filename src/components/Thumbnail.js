import React from "react";
import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles(() => ({
    thumbnail: {
        minWidth: "325px",
        width: "35vw",
        display: "inline-block",
    },
    flex: {
        display: "flex",
        justifyContent: "center",
    },
    overlay: {
        position: "relative",
        display: "inline-block",
        zIndex: 1,
    },
    icon: {
        position: "absolute",
        left: "40%",
        top: "40%",
        display: "block",
        transform: "translate(-40%, -42.5%)",
    },
    blur: {
        minWidth: "325px",
        width: "35vw",
        display: "inline-block",
        filter: "blur(2px)",
    },
}));

const Thumbnail = ({ file, progress }) => {
    const classes = useStyles();

    if (!file) {
        return null;
    }

    let thumbnail = URL.createObjectURL(file);

    return (
        <div>
            <p>Preview:</p>
            <div className={classes.flex}>
                <img
                    className={progress ? classes.blur : classes.thumbnail}
                    src={thumbnail}
                    alt="Preview of uploaded comic"
                ></img>
            </div>
        </div>
    );
};

export default Thumbnail;
