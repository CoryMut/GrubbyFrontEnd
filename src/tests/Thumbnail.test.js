import React from "react";
import { shallow } from "enzyme";
import Thumbnail from "../components/Thumbnail";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";

jest.mock("react-router-dom", () => ({
    useNavigate: () => ({
        push: jest.fn(),
    }),
    useLocation: () => ({
        push: jest.fn(),
    }),
}));

test("should test NavBar component", () => {
    const wrapper = shallow(<Thumbnail />);
    expect(wrapper).toMatchSnapshot();
});

test("it renders without crashing", () => {
    render(
        <MemoryRouter>
            <Thumbnail />
        </MemoryRouter>
    );
});
