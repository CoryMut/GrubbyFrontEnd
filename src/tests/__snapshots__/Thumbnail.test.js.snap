// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test NavBar component 1`] = `""`;
